medallion/views/objects.py

```diff
@@ -50,6 +49,102 @@ def get_custom_headers(api_root, id_):
     return headers
 
 
+def get_and_enforce_limit(api_root, id_, objects):
+    """
+    Defines TAXII API - Pagination:
+    Pagination section (3.5 <link here>`__), Get Object Manifests section (5.3 <link here>`__), and Get Objects section (5.4 <link here>`__)
+
+    Args:
+        api_root (str): the base URL of the API Root
+        id_ (str): the `identifier` of the Collection being requested
+        objects (dict): objects being returned for request (may have been filtered by other query parameters)
+
+    Returns:
+        headers:
+            header values matching objects being sent in server response
+        objects:
+            Values matching server request, sorted by ascending date_added
+
+    """
+    headers = {}
+    if request.args.get('limit'):
+        limit = int(request.args['limit'])
+    else:
+        limit = current_app.taxii_config["max_page_size"]
```

Comment:
The logic here is still off. This section of the code is saying that when we don't use the `limit` parameter we should use the server configured maximum. This is not the case, if limit is not provided in a request we always return the whole result.

The max_page_size is only meant to override the provided limit if its bigger than the server maximum.